runtime/src/main/java/org/corfudb/runtime/clients/BaseClient.java

```diff
@@ -56,55 +41,52 @@ public boolean pingSync() {
-        log.info("sealRemoteServer: send SEAL from me(clientId={}) to new epoch {}",
-                msg.getClientID(), epoch);
-        return router.sendMessageAndGetCompletable(msg);
-    }
-
-    public CompletableFuture<VersionInfo> getVersionInfo() {
-        return router.sendMessageAndGetCompletable(
-                new CorfuMsg(CorfuMsgType.VERSION_REQUEST).setEpoch(epoch).setClusterID(clusterId));
+    public CompletableFuture<Boolean> ping() {
+        return sendRequestWithFuture(getPingRequestMsg(), true, true);
     }
 
-
     /**
-     * Ping the endpoint, asynchronously.
+     * Restart the endpoint, asynchronously.
      *
      * @return A completable future which will be completed with True if
-     * the endpoint is reachable, otherwise False or exceptional completion.
+     * the endpoint restarts successfully, otherwise False or exceptional completion.
      */
-    public CompletableFuture<Boolean> ping() {
-        return router.sendMessageAndGetCompletable(
-                new CorfuMsg(CorfuMsgType.PING).setEpoch(epoch).setClusterID(clusterId));
+    public CompletableFuture<Boolean> restart() {
+        return sendRequestWithFuture(getRestartRequestMsg(), true, true);
     }
 
     /**
      * Reset the endpoint, asynchronously.
      * WARNING: ALL EXISTING DATA ON THIS NODE WILL BE LOST.
      *
      * @return A completable future which will be completed with True if
-     * the endpoint acks, otherwise False or exceptional completion.
+     * the endpoint resets successfully, otherwise False or exceptional completion.
      */
     public CompletableFuture<Boolean> reset() {
-        return router.sendMessageAndGetCompletable(new CorfuMsg(CorfuMsgType.RESET)
-                .setEpoch(epoch).setClusterID(clusterId));
+        return sendRequestWithFuture(getResetRequestMsg(), true, true);
     }
 
     /**
-     * Restart the endpoint, asynchronously.
+     * Sets the epoch on client router and on the target layout server.
      *
-     * @return A completable future which will be completed with True if
-     * the endpoint acks, otherwise False or exceptional completion.
+     * @param newEpoch New Epoch to be set
+     * @return Completable future which returns true on successful epoch set.
      */
-    public CompletableFuture<Boolean> restart() {
-        return router.sendMessageAndGetCompletable(new CorfuMsg(CorfuMsgType.RESTART)
-                .setEpoch(epoch).setClusterID(clusterId));
+    public CompletableFuture<Boolean> sealRemoteServer(long newEpoch) {
+        return sendRequestWithFuture(getSealRequestMsg(newEpoch), false, true);
+    }
+
+    /**
+     * Get the version info from the target layout server.
+     *
+     * @return Completable future which returns {@link VersionInfo} object.
+     */
+    public CompletableFuture<VersionInfo> getVersionInfo() {
+        return sendRequestWithFuture(getVersionRequestMsg(), true, true);
```

Comment:
Why is this (true, true), on master it is (true, false)

`SEAL(3, new TypeToken<CorfuPayloadMsg<Long>>() {}, true, false),`

Reply:
Never mind, the order is reversed.